src/dmd/toir.d

```diff
@@ -392,385 +392,115 @@ enum NotIntrinsic = -1;
-            //printf("test1 %s %s %d\n", core_namearray[i], core_namearray[i + 1], strcmp(core_namearray[i], core_namearray[i + 1]));
-            if (strcmp(core_namearray[i], core_namearray[i + 1]) >= 0)
-            {
-                printf("core_namearray[%ld] = '%s'\n", cast(long)i, core_namearray[i]);
-                assert(0);
-            }
+                 if (id3 == Id.volatileLoad)  op = OPind;
+            else if (id3 == Id.volatileStore) op = OPeq;
+
+            else if (id3 == Id.bsf) op = OPbsf;
+            else if (id3 == Id.bsr) op = OPbsr;
+            else if (id3 == Id.btc) op = OPbtc;
+            else if (id3 == Id.btr) op = OPbtr;
+            else if (id3 == Id.bts) op = OPbts;
+
+            else if (id3 == Id.inp)  op = OPinp;
+            else if (id3 == Id.inpl) op = OPinp;
+            else if (id3 == Id.inpw) op = OPinp;
+
+            else if (id3 == Id.outp)  op = OPoutp;
+            else if (id3 == Id.outpl) op = OPoutp;
+            else if (id3 == Id.outpw) op = OPoutp;
+
+            else if (id3 == Id.bswap)   op = OPbswap;
+            else if (id3 == Id._popcnt) op = OPpopcnt;
         }
-        for (size_t i = 0; i < core_namearray64.length - 1; i++)
+        else if (id2 == Id.volatile)
         {
-            if (strcmp(core_namearray64[i], core_namearray64[i + 1]) >= 0)
-            {
-                printf("core_namearray64[%ld] = '%s'\n", cast(long)i, core_namearray64[i]);
-                assert(0);
-            }
+                 if (id3 == Id.volatileLoad)  op = OPind;
+            else if (id3 == Id.volatileStore) op = OPeq;
         }
     }
 
-    size_t length = strlen(name);
-
-    if (length > 10 &&
-        (name[7] == 'm' || name[7] == 'i') &&
-        !memcmp(name, "_D3std".ptr, 6))
+    // No 64-bit bsf bsr btc in 32bit mode
+    if (!global.params.is64bit)
     {
-        int i = binary(name + 6,
-            cast(const(char)**)(global.params.is64bit ? std_namearray64.ptr : std_namearray.ptr),
-            cast(int)std_namearray.length);
-        return (i == -1) ? NotIntrinsic : std_ioptab[i];
-    }
-    if (length > 12 &&
-        (name[8] == 'm' || name[8] == 'b' || name[8] == 's' || name[8] == 'v') &&
-        !memcmp(name, "_D4core".ptr, 7))
-    {
-        int i = binary(name + 7,
-            cast(const(char)**)(global.params.is64bit ? core_namearray64.ptr : core_namearray.ptr),
-            cast(int)core_namearray.length);
-        if (i != -1)
-        {
-            int op = core_ioptab[i];
-            if (!global.params.is64bit &&
-                (op == OPbsf || op == OPbsr) &&
-                op == core_ioptab[i - 1])
-            {
-                // Don't recognize 64 bit bsf() / bsr() in 32 bit mode
-                op = NotIntrinsic;
-            }
-            return op;
+        if (op == OPbsf || op == OPbsr ) {
+            //_D4core5bitop3bsfFNaNbNiNfxZi
+            // index of x == 26, m == ulong
+            if (mangleExact(fd)[26] == 'm')
```

Comment:
If the attributes of `core.bitop.bsf` change, this might read garbage.

Reply:
Yeah I don't like that either, but its much faster than a `strcmp` and for some reason I couldn't get a type check on `fd` to do what I wanted: `const bool isFirstArgUlong = fd.parameters && (*fd.parameters)[0].type == Type.tuns64;`

Reply:
This PR already avoids a lot of string comparisons, so I think adding one is fine here. You could implement a small local `endsWith` to skip the attribute checks.